src/ReactiveSocket.cpp

```diff
@@ -75,6 +76,20 @@ void ReactiveSocket::requestSubscription(
   automaton->start();
 }
 
+void ReactiveSocket::requestFireAndForget(
```

Comment:
Why does this method allow caller to provide a Subscriber? What that subscriber mean and what kind of data does it expect? Since the signatures of requestFNF and requestSubscription are the same, what is the difference between them?

I wouldn't be concerned by compatibility with Java. ReactiveStreams are not "compatible", because they need to work in a language without a GC. ReactiveSocket intentionally has control flow inversion, as that limits laziness and makes memory management easier and allows to keep more stuff on the stack. Most important to me is for this code to be idiomatic (in C++).